pkg/controllers/routercerts/controller.go

```diff
@@ -207,6 +207,13 @@ func (c *routerCertsDomainValidationController) validateRouterCertificates() ope
 		return newRouterCertsDegradedf("InvalidServerCertRouterCerts", "secret/%v.spec.data[%v] -n %v: certificate could not validate route hostname %v: %v", c.defaultSecretName, ingressDomain, c.secretNamespace, verifyOptions.DNSName, err)
 	}
 
+	// check the server certificates for reliance on the deprecated CN field
```

Comment:
This is a redundant change, `verifyWithAnyCertificate` should already return errors.

You should be looking at https://github.com/openshift/cluster-authentication-operator/blob/55b274c54c4a404799dea0e9e41f1b62e9f84c55/pkg/controllers/customroute/custom_route_controller.go

Reply:
Until Go v1.17, the `GODEBUG=x509ignoreCN=0` flag (which is embedded in all base images) will prevent `verifyWithAnyCertificate`, or any other certificate parsing, from failing due to missing SAN fields. This patch is targeted against OCP v4.9 which is compiled with Go v1.16.

Reply:
That being said, `custom_route_controller` seems like a good tip. Do you think that the check should be implemented in both places? @stlaz

Reply:
Oh and thank you for the review!

Reply:
I have added the check in `ValidateServerCert`, which is the validation function called in the custom route controller loop. Thank you!

@stlaz PTAL

Reply:
I suppose we could have this check even here, although the description should say: "open a bug to the Routing component". cc @openshift/openshift-team-network-edge

Reply:
@stlaz i believe we had a discussion round with the router team. their assertion is that they don't provision the cert per se, so it is an exercise on the corresponding operators to do the validation.

Reply:
Thank you. I understand that this means that we keep both checks in place.

Reply:
> we keep both checks in place.

agreed, hence SAN-less certs can happen both in the central route cert and in custom route certs and for both cases we should verify.

Reply:
> @stlaz i believe we had a discussion round with the router team. their assertion is that they don't provision the cert per se, so it is an exercise on the corresponding operators to do the validation.

I believe they eventually decided to introduce the checks. If they did not check their inputs, our operators would go degraded.